src/ui/reducers/protocolMessages.ts

```diff
@@ -83,7 +84,7 @@ const protocolMessagesSlice = createSlice({
           }
           case "error": {
             const { value: error } = message;
-            state.idToErrorMap[error.id] = error;
+            state.idToErrorMap[error.id] = cloneDeep(error);
```

Comment:
This makes sense.

Is it possible this could cause problems anywhere that may rely on the data to stay in-sync after a mutation? (Do we read source stuff from Redux and use it for the related sources logic? Or does that always come from Suspense caches?

Reply:
This redux slice is only used for the protocol viewer, I don't think it expects any updates.

> Do we read source stuff from Redux and use it for the related sources logic?

We're still using the sources from redux in some places but that is independent of this redux slice.

Reply:
Would probably be good for us to stop putting this data in Redux and just use our suspense caches directly (since it would avoid the potential for being out of sync here)